### PR TITLE
buildsystem: sane warning if not using gmake

### LINKED ISSUE #952: Add warning if using `make` instead of `gmake`
In the main GAP directory, there's the following:

https://github.com/gap-system/gap/blob/master/Makefile

which prints:

~~~
Please use GNU make to build GAP (try 'gmake' or 'gnumake')
*** Error 1 in /home/jdm/gap (Makefile:22 '.DEFAULT': @exit 1)
~~~

if one uses `make` instead of `gmake`. Whereas for Semigroups we get the less helpful:

~~~
*** Parse error in /home/jdm/gap/pkg/Semigroups: Need an operator in 'HPCOMBI_CONSTEXPR_FUN_ARGS' (Makefile:36)
*** Parse error: Need an operator in 'endif' (Makefile:38)
*** Parse error: Need an operator in 'LIBSEMIGROUPS_HPCOMBI_ENABLED' (Makefile:40)
*** Parse error: Need an operator in 'endif' (Makefile:42)
*** Parse error: Need an operator in 'WITH_INCLUDED_LIBSEMIGROUPS' (Makefile:44)
*** Parse error: Need an operator in 'LIBSEMIGROUPS_HPCOMBI_ENABLED' (Makefile:48)
*** Parse error: Need an operator in 'endif' (Makefile:51)
*** Parse error: Need an operator in 'endif' (Makefile:54)
*** Parse error: Need an operator in 'KERNEL_DEBUG' (Makefile:57)
*** Parse error: Need an operator in 'else' (Makefile:59)
*** Parse error: Need an operator in 'endif' (Makefile:61)
*** Parse error: Need an operator in 'SYS_IS_CYGWIN' (Makefile:72)
*** Parse error: Need an operator in 'endif' (Makefile:74)
*** Parse error: Need an operator in 'GAP_KERNEL_MAJOR_VERSION' (Makefile.gappkg:42)
*** Parse error: Need an operator in 'endif' (Makefile.gappkg:45)
*** Parse error: Need an operator in 'KEXT_USE_AUTOCONF' (Makefile.gappkg:67)
*** Parse error: Need an operator in 'else' (Makefile.gappkg:69)
*** Parse error: Need an operator in 'endif' (Makefile.gappkg:71)
*** Parse error: Missing dependency operator (Makefile.gappkg:93)
*** Parse error: Need an operator in 'V' (Makefile.gappkg:94)
*** Parse error: Need an operator in 'endif' (Makefile.gappkg:96)
*** Parse error: Need an operator in 'endif' (Makefile.gappkg:97)
*** Parse error: Filename missing from "-include" (Makefile.gappkg:108)
*** Parse error: Need an operator in 'KEXT_USE_AUTOCONF' (Makefile.gappkg:170)
*** Parse error: Missing dependency operator (Makefile.gappkg:175)
*** Parse error: Missing dependency operator (Makefile.gappkg:183)
*** Parse error: Missing dependency operator (Makefile.gappkg:202)
*** Parse error: Missing dependency operator (Makefile.gappkg:213)
*** Parse error: Missing dependency operator (Makefile.gappkg:215)
*** Parse error: Need an operator in 'WITH_INCLUDED_LIBSEMIGROUPS' (Makefile:83)
*** Parse error: Need an operator in 'endif' (Makefile:96)
~~~

It'd be great if we could have a nice warning instead of the less informative above.

### PR DESCRIPTION
**Pull request by james-d-mitchell**
This is an attempt to follow 

https://github.com/semigroups/Semigroups/pull/947#issuecomment-1718015553

so that a sane warning is emitted if trying to use `make` instead of `gmake`. Hopefully resolves #952. Thanks @fingolfin for the hints!

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#
+# This Makefile serves two purposes:
+# 1) If the user types "make" without having run "configure", we suggest running configure.
+# 2) Our build system is written for GNU make, and makes liberal users of its features.
```
Comment (fingolfin) [inline Makefile:4]:
```suggestion
# 2) Our build system is written for GNU make, and makes liberal use of its features.
```

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+#    We therefore put it into "GNUmakefile", which is picked up by GNU make, but ignore by
+#    other make versions, such as BSD make.
+#    Thus, if the user has BSD make, it will run this Makefile instead -- and we inform
+#    them that they need to use GNU make to compile GAP.
+#
+# To learn more about the GAP build system, see README.buildsys.md
+#
+.DEFAULT:
+	@if test -f GNUmakefile ; then \
+	    printf "Please use GNU make to build GAP (try 'gmake' or 'gnumake')\n" ; \
```
Comment (fingolfin) [inline Makefile:14]: Again says "GAP"

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,23 @@
+#
+# This Makefile serves two purposes:
+# 1) If the user types "make" without having run "configure", we suggest running configure.
+# 2) Our build system is written for GNU make, and makes liberal users of its features.
+#    We therefore put it into "GNUmakefile", which is picked up by GNU make, but ignore by
+#    other make versions, such as BSD make.
+#    Thus, if the user has BSD make, it will run this Makefile instead -- and we inform
+#    them that they need to use GNU make to compile GAP.
```
[the remark this answers is not in the document]
Comment (james-d-mitchell) [reply Makefile]: Thanks @fingolfin ! Done.

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,23 @@
+#
+# This Makefile serves two purposes:
+# 1) If the user types "make" without having run "configure", we suggest running configure.
+# 2) Our build system is written for GNU make, and makes liberal users of its features.
+#    We therefore put it into "GNUmakefile", which is picked up by GNU make, but ignore by
```
[the remark this answers is not in the document]
Comment (james-d-mitchell) [reply Makefile]: I guess these typos are also in the GAP repo version of this file, or I they resolved and I haven't pulled the changes yet.

**Comment by james-d-mitchell** [comment]
This seems to work on Mac OSX, and has the desired behaviour on openbsd too (where I first noticed the issue that this PR resolves). Will merge if the CI passes ok too.

**Pull request merged**